src/ui/favoritespanewidget.cc

```diff
@@ -923,6 +989,54 @@ TreeItem * FavoritesModel::getItem( const QModelIndex & index ) const
   return rootItem;
 }
 
+TreeItem * FavoritesModel::getItemByFullPath( const QStringList & fullPath ) const
+{
+  TreeItem * targetFolder = nullptr;
+  TreeItem * parentItem   = getItem( QModelIndex() );
+  for ( qsizetype fullPathPos = 0; fullPathPos < fullPath.count(); ++fullPathPos ) {
+    for ( int i = 0; i < parentItem->childCount(); i++ ) {
+      TreeItem * item = parentItem->child( i );
+      if ( item->type() == TreeItem::Folder ) {
+        if ( item->data().toString() == fullPath[ fullPathPos ] ) {
+          if ( fullPathPos == ( fullPath.count() - 1 ) ) {
+            targetFolder = item;
+          }
+          else {
+            parentItem = item;
+            break;
+          }
+        }
+      }
+    }
+  }
```

Comment:
need optimization. and seems not correct.

for example ,   if qstringlist is ('a','b','c')
the parentItem's child is `d`  ,there is no need to continue.

Reply:
fix this , then we can merge it .